src/domain/task/utils/task.utils.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { progressInvariant } from './task.utils'
+
+describe('Considering the progressInvariant() function', () => {
+  describe.each`
+    current      | min          | max          | expectedResult
+    ${undefined} | ${undefined} | ${undefined} | ${true}
+    ${10}        | ${undefined} | ${undefined} | ${true}
```

Comment:
What's the meaning of this case? You can specify a progress value without having defined the bounds?

Reply:
Yep, as all properties are optional, I think it's kind of weird to prevent from setting a current value without bounds.
Maybe I'm wrong, but I think we may meet some use cases where we have to set initial current value before bounds.
If it's OK for you, I will preserve this behavior 😌